doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+        /// CatalogPackage object from a manifest.
+        /// DESIGN NOTE: This is proposed as the preferred alternative to having InstallOptions match the winget
+        /// command line which allows you to pass in a manifestPath directly into install. Allowing creation of 
+        /// a CatalogPackage object from a manifest allows the install api to consistently use package objects 
+        /// rather than having a "you can either set the manifest path or use a package but not both" problem.
+        static CatalogPackage TryCreateFromManifest(String manifestPath);
+
+        /// IMPLEMENTATION NOTE: PackageProperty fields from AppInstallerRepositorySearch.h
+        /// Gets a property of this package.
+        String Id { get; };
+        String Name { get; };
+
+        /// Gets the installed package information if the package is installed.
+        PackageVersionInfo InstalledVersion{ get; };
+
+        /// Gets all available versions of this package.
+        /// The versions will be returned in sorted, descending order.
+        ///  Ex. { 4, 3, 2, 1 }
+        Windows.Foundation.Collections.IVectorView<PackageVersionId> AvailableVersions { get; };
+
+        /// Gets a specific version of this package.
+        PackageVersionInfo LatestAvailableVersion { get; };
+
+        /// Gets a specific version of this package.
+        PackageVersionInfo GetAvailableVersion(PackageVersionId versionKey);
+
+        /// Gets a value indicating whether an available version is newer than the installed version.
+        Boolean IsUpdateAvailable { get; };
+
+        /// Gets whether the package is installing
+        Boolean IsInstalling { get; };
+
+        /// DESIGN NOTE:
+        /// IsSame from IPackage in AppInstallerRepositorySearch is not implemented in V1.
+        /// Determines if the given IPackage refers to the same package as this one.
+        /// virtual bool IsSame(const IPackage*) const = 0;
+    }
+
+    /// IMPLEMENTATION NOTE: CompositeSearchBehavior from AppInstallerRepositorySource.h
+    /// Search behavior for composite sources.
+    /// Only relevant for composite sources with a local source, not for aggregates of multiple available sources.
+    /// Installed and available packages in the result are always correlated when possible.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum CompositeSearchBehavior
+    {
+        /// Search only installed packages
+        InstalledPackages,
+        /// Search only installing packages
+        InstallingPackages,
+        /// Search only local packages (installed and installing)
+        AllLocalPackages,
+        /// Search both local and remote sources.
+        AllPackages,
+    };
+
+    /// IMPLEMENTATION NOTE: MatchType from AppInstallerRepositorySearch.h
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum MatchType
+    {
+        Exact,
+        CaseInsensitive,

```

Comment:
Which locale's case-insensitivity map do we use? The locale of the OS? The locale of the user? The locale of the app issuing the search? The locale of the app being searched for? (So searching for "LIRA" would not match a Turkish app called "Lira" but would match an Italian app called "Lira".) The invariant locale?

Are other locale-sensitive equivalences considered beyond case? Would "Fuenf" match "Fünf" in Germany?

Reply:
DISCUSSION: Service-backed implementations will do whatever they do. Local catalogs use ICU case folding on a normalized string.

RECOMMEND: Validate with @shawnsteele for details